src/deepsparse/transformers/pipelines/embedding_extraction.py

```diff
@@ -0,0 +1,438 @@
+        """
+        onnx_path = super().setup_onnx_file_path()
+
+        if self._emb_extraction_layer is not None or self._final_node_name is not None:
+            (
+                onnx_path,
+                self.onnx_output_names,
+                self._temp_model_directory,
+            ) = truncate_transformer_onnx_model(
+                onnx_path,
+                emb_extraction_layer=self._emb_extraction_layer,
+                hidden_layer_size=self._model_size,
+                final_node_name=self._final_node_name,
+            )
+        else:
+            _LOGGER.info("Skipping model truncation")
+
+        return onnx_path
+
+    def parse_inputs(self, *args, **kwargs) -> BaseModel:
+        """
+        :param args: ordered arguments to pipeline, either a input_schema object,
+            a string text, or a list of inputs
+        :param kwargs: keyword arguments to pipeline
+        :return: pipeline arguments parsed into the given `input_schema`
+            schema if necessary. If an instance of the `input_schema` is provided
+            it will be returned
+        """
+        if args and kwargs:
+            raise ValueError(
+                f"{self.__class__} only support args OR kwargs. Found "
+                f" {len(args)} args and {len(kwargs)} kwargs"
+            )
+
+        if args:
+            if isinstance(args, str):
+                return self.input_schema(inputs=[args[0]])
+
+            if len(args) == 1:
+                if isinstance(args[0], self.input_schema):
+                    return args[0]
+                else:
+                    return self.input_schema(inputs=args[0])
+            else:
+                return self.input_schema(inputs=args)
+        else:
+            return self.input_schema(**kwargs)
+
+    def process_inputs(self, inputs: EmbeddingExtractionInput) -> List[numpy.ndarray]:
+        """
+        Tokenizes input
+
+        :param inputs: inputs to the pipeline.
+        :return: inputs of this model processed into a list of numpy arrays that
+            can be directly passed into the forward pass of the pipeline engine
+        """
+        if isinstance(inputs.inputs, str):
+            inputs.inputs = [inputs.inputs]
+
+        tokens = self.tokenizer(
+            inputs.inputs,
+            add_special_tokens=True,
+            return_tensors="np",
+            padding=PaddingStrategy.MAX_LENGTH.value,
+            truncation=TruncationStrategy.LONGEST_FIRST.value,
+        )
+
+        # mask padding and cls_token
+        pad_masks = tokens["input_ids"] == self.tokenizer.pad_token_id
+        cls_masks = tokens["input_ids"] == self.tokenizer.cls_token_id
+
+        return self.tokens_to_engine_input(tokens), {
+            "pad_masks": pad_masks,
+            "cls_masks": cls_masks,
+        }
+
+    def engine_forward(self, engine_inputs: List[numpy.ndarray]) -> List[numpy.ndarray]:
```

Comment:
we likely don't have to/shouldn't be overriding engine_forward for this pipeline